vault/resource_namespace.go

```diff
@@ -142,18 +193,27 @@ func namespaceRead(d *schema.ResourceData, meta interface{}) error {
 	d.SetId(resp.Data[consts.FieldPath].(string))
 
 	toSet := map[string]interface{}{
-		consts.FieldNamespaceID: resp.Data["id"],
+		consts.FieldNamespaceID: resp.Data[consts.FieldID],
 		consts.FieldPath:        util.TrimSlashes(path),
 	}
 
+	if provider.IsAPISupported(meta, provider.VaultVersion112) {
+		toSet[consts.FieldCustomMetadata] = resp.Data[consts.FieldCustomMetadata]
+	} else {
+		// set computed parameter to nil for vault versions <= 1.11
+		// prevents 'known after apply' drift in TF state since field
+		// would never be set otherwise
+		toSet[consts.FieldCustomMetadata] = nil
```

Comment:
nit: you could probably add the key with the nil value in the `toSet` map declaration.